spring-session/src/main/java/org/springframework/session/data/redis/RedisOperationsSessionRepository.java

```diff
@@ -696,7 +708,9 @@ static String getSessionAttrNameKey(String attributeName) {
 		RedisSession(MapSession cached) {
 			Assert.notNull("MapSession cannot be null");
 			this.cached = cached;
-			this.originalPrincipalName = PRINCIPAL_NAME_RESOLVER.resolvePrincipal(this);
+			Map<String, String> indexes = RedisOperationsSessionRepository.this
+					.indexResolver.resolveIndexesFor(this);
```

Comment:
This would benefit from `IndexResolver` having a method to resolve single index.